File: Python3/Medium/Accepted/maximum-number-of-vowels-in-a-substring-of-given-length_1.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        
        max_vowels = 0
        vowels = 0
        for i in range(len(s)):
            if s[i] in "aeiou":
                vowels += 1
            if i >= k and s[i-k] in "aeiou":
                vowels -= 1
            max_vowels = max(max_vowels, vowels)
        return max_vowels

File: Rust/Medium/TimeLimitExceeded/maximum-number-of-vowels-in-a-substring-of-given-length_1.rs
impl Solution {
    pub fn max_vowels(s: String, k: i32) -> i32 {
        
        let k = k as usize;
        let mut res = 0;
        for i in 0..k {
            if "aeiou".contains(s.chars().nth(i).unwrap()) {
                res += 1;
            }
        }
        let mut cur = res;
        for i in k..s.len() {
            if "aeiou".contains(s.chars().nth(i).unwrap()) {
                cur += 1;
            }
            if "aeiou".contains(s.chars().nth(i - k).unwrap()) {
                cur -= 1;
            }
            res = res.max(cur);
        }
        res
    }
}